pkgs/development/compilers/julia/1.0-bin.nix

```diff
@@ -0,0 +1,72 @@
+{ autoPatchelfHook, fetchurl, lib, makeWrapper, openssl, stdenv }:
+
+stdenv.mkDerivation rec {
+  pname = "julia-bin";
+  version = "1.0.5";
+
+  src = {
+    x86_64-linux = fetchurl {
+      url = "https://julialang-s3.julialang.org/bin/linux/x64/${lib.versions.majorMinor version}/julia-${version}-linux-x86_64.tar.gz";
+      sha256 = "00vbszpjmz47nqy19v83xa463ajhzwanjyg5mvcfp9kvfw9xdvcx";
+    };
+  }.${stdenv.hostPlatform.system} or (throw "Unsupported system: ${stdenv.hostPlatform.system}");
+
+  # Julia’s source files are in different locations for source and binary
+  # releases. Thus we temporarily create symlinks to allow us to share patches
+  # with source releases.
+  prePatch = ''
+    ln -s share/julia/stdlib/v${lib.versions.majorMinor version} stdlib
+    ln -s share/julia/test
+  '';
+  patches = [
+    # Source release Nix patch(es) relevant for binary releases as well.
+    ./patches/1.0-bin/0002-nix-Skip-tests-that-require-network-access.patch
+  ];
+  postPatch = ''
+    # Revert symlink hack.
+    rm stdlib test
+  '';
+
+  buildInputs = [ makeWrapper ];
+  nativeBuildInputs = [ autoPatchelfHook ];
+
+  installPhase = ''
+    runHook preInstall
+    cp -r . $out
+    # Setting `LD_LIBRARY_PATH` resolves `Libdl` failures. Not sure why this is
+    # only necessary on v1.0.x and a cleaner solution is welcome, but after
+    # staring at `strace` for a few hours this is as clean as I could make it.
```

Comment:
Maybe 1.0 adds `$out/lib` to `RPATH` but links no libraries (using `dlopen` instead), so Nixpkgs automatic patchelf removes it from `RPATH`?

Reply:
Thank you, that could be it. Will investigate.

Reply:
Yep, I suspect you are bang on:

```bash
> objdump -x lib/libjulia.so | grep RPATH
  RPATH                $ORIGIN/julia:$ORIGIN
> objdump -x /nix/store/pxs6qkddkhpvk1hpkahgz4w18ahfd3m1-julia-bin-1.0.5/lib/libjulia.so | grep RPATH
>
```

Digging through the manual I thought that what I should do was to set `dontPatchELF` to true, but I still end up with the RPATH being removed.

Reply:
If this is not easily addressable, I am happy to merge this as is and try to clean it up later.